src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/DateTimeOffsetConverter.shared.cs

```diff
@@ -1,43 +1,30 @@
 ﻿using System;
-using Xamarin.CommunityToolkit.Extensions.Internals;
-using Xamarin.Forms;
 
 namespace Xamarin.CommunityToolkit.Converters
 {
 	/// <summary>
 	/// Converts <see cref="DateTimeOffset"/> to <see cref="DateTime"/> and back.
 	/// </summary>
-	public class DateTimeOffsetConverter : ValueConverterExtension, IValueConverter
+	public class DateTimeOffsetConverter : BaseConverter<DateTimeOffset, DateTime>
 	{
 		/// <summary>
-		/// Converts <see cref="DateTimeOffset"/> to <see cref="DateTime"/>
+		/// Converts <see cref="DateTime"/> back to <see cref="DateTimeOffset"/>.
 		/// </summary>
 		/// <param name="value">The value to convert.</param>
-		/// <param name="targetType">The type of the binding target property. This is not implemented.</param>
-		/// <param name="parameter">Additional parameter for the converter to handle. This is not implemented.</param>
-		/// <param name="culture">The culture to use in the converter. This is not implemented.</param>
-		/// <returns>The <see cref="DateTime"/> value.</returns>
-		public object Convert(object? value, Type targetType, object? parameter, System.Globalization.CultureInfo culture)
-			=> value is DateTimeOffset dateTimeOffset
-				? dateTimeOffset.DateTime
-				: throw new ArgumentException("Value is not a valid DateTimeOffset", nameof(value));
+		/// <returns>The <see cref="DateTimeOffset"/> value.</returns>
+		public override DateTimeOffset ConvertBackTo(DateTime value) =>
+			value.Kind switch
+			{
+				DateTimeKind.Local => new DateTimeOffset(DateTime.SpecifyKind(value, DateTimeKind.Unspecified), DateTimeOffset.Now.Offset),
```

Comment:
I am open to anyone and everyone reviewing this change. I believe this solves our failing unit test when running locally based on this documentation: https://stackoverflow.com/a/36255983/32348

I suspect it passed on the CI as the agents are likely running in UTC and therefore the local test is no different to the UTC test.

Potentially also solves https://github.com/xamarin/XamarinCommunityToolkit/issues/1744

@Countryen if you grab the latest changes does the DateTimeOffsetConverter_Tests all pass now?

Reply:
@bijington  Yes, all (including all DateTimeOffsetConverter_Tests) tests now pass green for me ✔️